### PR TITLE
feat(preferences): added preferences and toggles for tracks and challenges

### PR DESCRIPTION
**Pull request by NickBurnett**
…enges

### CODE REVIEW

**Comment by NickBurnett** [comment]
ok you can ignore the description at the top idk wtf happened

File: pages/api/preferences/index.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    return res.status(200).json({
+      code: 'set-preferences',
+      message: `Preferences updated.`,
+      data: { ...toggles },
+    });
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({
+      code: 'internal-error',
+      message: 'Something went wrong when processing this request. Tell Nick and he will fix it.',
```
Comment (abdullah50053) [inline pages/api/preferences/index.tsx:82]: hmm

**Review by abdullah50053 — APPROVED** [review]

**Review by abdullah50053 — CHANGES_REQUESTED** [review]
Current implementation does not hide challenges under `Dashboard/HackCenter`

**Review by abdullah50053 — APPROVED** [review]
l g t m

**Pull request merged**